paymentsheet/src/main/java/com/stripe/android/paymentsheet/BaseAddPaymentMethodFragment.kt

```diff
@@ -101,11 +105,18 @@ internal abstract class BaseAddPaymentMethodFragment : Fragment() {
         sheetViewModel.eventReporter.onShowNewPaymentOptionForm()
     }
 
+    override fun onStop() {
+        super.onStop()
+        composeFragmentFieldValues?.cancel()
```

Comment:
This feels like we may not be scoping the job correctly. Is it possible to scope the lifecycle of the job to this fragment rather than manually canceling the job?

Reply:
@awush-stripe I moved the BaseSheetViewModel so it is accessible from the ComposeFormDataCollectionFragment.  It was slightly akward as the VIewModel could be a PaymentSheetActivityViewModel or a PaymentOptionsActivityViewModel.  It seems to work calling activityViewModels from onViewCreated, but would be curious to hear if there was a better way to handle this.